From 1eb23b4b01728d5f828b42472c9639b5d9d43ab2 Mon Sep 17 00:00:00 2001
From: Alex Williamson <alex.williamson@redhat.com>
Date: Fri, 1 Aug 2014 18:12:52 -0500
Subject: [CHANGE 1/5] Add init_pci_busdevfn
To: rhvirt-patches@redhat.com,
    jen@redhat.com

Message-id: <20140513195620.6596.3782.stgit@bling.home>
Patchwork-id: 58832
O-Subject: [RHEL6.6 gpxe PATCH 1/5] [romprefix] Add init_pci_busdevfn
Bugzilla: 1057249
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Bugzilla: 1057249
Upstream commits: 132c391712b19fd76847f192d7523cf67d44f67f (ipxe)
                  cb37d92ff6915b511163e0160b5587cce4d9bb1c (ipxe)

This variable simply stores a register defined to contain the PCI
bus:dev.fn of the boot device.  This was added upstream in a much
larger commit, the first referenced above.  Since that deals with
with a much more comprehensive change and we only need a tiny part
if it, I've pulled it out here.  From the second commit, I'm simply
taking the change of the initial value.

Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
Signed-off-by: jen <jen@redhat.com>
---
 src/arch/i386/prefix/romprefix.S | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/arch/i386/prefix/romprefix.S b/src/arch/i386/prefix/romprefix.S
index 7d53237..f2a8618 100644
--- a/src/arch/i386/prefix/romprefix.S
+++ b/src/arch/i386/prefix/romprefix.S
@@ -166,6 +166,9 @@ init:
 	movw	%bx, %gs
 	movw	%di, %bx
 
+	/* Store PCI bus:dev.fn address */
+	movw	%ax, init_pci_busdevfn
+
 	/* Print message as early as possible */
 	movw	$init_message, %si
 	xorw	%di, %di
@@ -463,6 +466,13 @@ init_message_done:
 	.asciz	"\n\n"
 	.size	init_message_done, . - init_message_done
 
+/* PCI bus:dev.fn
+ *
+ */
+init_pci_busdevfn:
+	.word	0
+	.size	init_pci_busdevfn, . - init_pci_busdevfn
+
 /* ROM image location
  *
  * May be either within option ROM space, or within PMM-allocated block.
-- 
1.9.3

